fastify.js

```diff
@@ -293,7 +297,9 @@ function build (options) {
 
     req.log.info({ req }, 'incoming request')
 
-    var request = new context.Request(params, req, urlUtil.parse(req.url, true).query, req.headers, req.log)
+    var queryPrefix = req.url.indexOf('?')
+    var query = queryPrefix !== -1 ? querystringParser(req.url.slice(queryPrefix + 1)) : {}
```

Comment:
To make sure the type of the `query` object remains consistent, I think it would make sense for the `querystringParser` function to always be called. For example, `querystring.parse()` returns an object with a null prototype (`Object.create(null)`). A user might also want to use [`URLSearchParams`](https://nodejs.org/dist/latest/docs/api/url.html#url_class_urlsearchparams), which creates a map-like object, so it would be very inconsistent for `req.query` to be a either a `URLSearchParams` object or a plain object for different requests.